enhancements/windows-containers/csi-proxy.md

```diff
@@ -132,34 +132,111 @@ No action required.
+New cluster installs on other platforms should be created with a minumum version of OCP 4.12, so they can immediately
+use CSI driver storage, as there will not be a way to safely migrate in-tree volumes for those platforms. This is due
+to a lack of Windows in-tree storage support leading up to 4.12 for those platforms.
+
+##### Node upgrades
+Cluster upgrades will have an impact on this feature when upgrading from a cluster version which uses in-tree
+storage, to one which uses CSI drivers. [When upgrading cluster versions, in-tree driver Volumes will automatically
+be updated to use the appropriate CSI node drivers by the cluster](https://docs.openshift.com/container-platform/4.12/storage/container_storage_interface/persistent-storage-csi-migration.html).
+Volumes attached to Windows Nodes, will remain usable until the Node's kubelet is upgraded to the version matching the
+new cluster version. This overlap period will allow for a smooth transition for each Node.
+
+In order to properly make use of this, and ensure a clean transition with minimal workload interruption, both csi-proxy
+and the CSI node drivers must be running on the Windows node before the upgrade commences. It will be up to the cluster
+administrator to install the CSI drivers on the cluster. They should check that all deployed pods go into running, and
+a CSINode is created for each Windows node. The CSINode should list the CSI driver as enabled.
+
+The requirement of csi-proxy is one that must be fufilled by WMCO. As this feature is targeted for WMCO 9.0.0,
+csi-proxy must be backported to WMCO v7 and v8, to enable a smooth transition of volumes when upgrading from OCP
+4.12 and 4.13. This specifically effects Azure, which loses in-tree storage functionality in 4.13, and vSphere which
+loses it in 4.14.
+
+What this means for the user, is that when they are upgrading a cluster from a version that uses in-tree to one that
+uses CSI, they must first upgrade WMCO to a version which runs csi-proxy on Windows Nodes. Only then should they
+upgrade the cluster.
+
+In order to help ensure a smooth transition WMCO v8 and v9 will block each individual Node upgrade if:
+1) A Node is being upgraded from in-tree storage to CSI driver storage
+2) A volume is attached to the Node
+
+The cluster administrator will need to apply the label `windowsmachineconfig.openshift.io/allow-upgrade=true` to each
+blocked Node, in order to unblock the upgrade. This way they can unblock an upgrade, and ensure that the workload is
+able to go back into running, using the migrated storage volume. If the workload does not go back into running, the
+cluster admin must debug any issues with the CSI drivers, and correct them, before allowing upgrades on any other
+Nodes. This is meant to minimize any workload interruption on upgrade.
+
+##### Possible upgrade path scenarios
+
+###### v7 -> v8
+
+Scenario A:
+
+v7 with csi-proxy backport -> v8 with csi-proxy backport
```

Comment:
> v7 with csi-proxy backport

Instead of `backport` I recommend sticking to the wording used above: 

```
v7 with csi-proxy running as a Windows service
```